### PR TITLE
Refactored code, according to dp-import-cantabular-dataset standard

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

Refactored code to simplify service package, and a few other optimizations.

### How to review

- Make sure code changes make sense (you can compare them with dp-import-cantabular-dataset)
- Make sure unit and component tests pass
- You may run this service along with the producer and validate that the messages are correctly consumed.

### Who can review

Anyone

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,29 +27,38 @@ func main() {
...
 	svcErrors := make(chan error, 1)
-	svcList := service.NewServiceList(&service.Init{})
-	svc, err := service.Run(ctx, svcList, BuildTime, GitCommit, Version, svcErrors)
+
+	// Read config
+	cfg, err := config.Get()
 	if err != nil {
-		return errors.Wrap(err, "running service failed")
+		log.Error(ctx, "unable to retrieve configuration", err)
+		return err
```
Comment (ChrysmOre) [inline main.go:44]: Looks like we should be returning fmt.Errorf wrapped error here? Maybe I need to update on import-cantabular-dataset too

File: main.go [hunk 2]
```diff
@@ -26,29 +27,38 @@ func main() {
...
 	}
+	svc.Start(ctx, svcErrors)
 
 	// blocks until an os interrupt or a fatal error occurs
 	select {
 	case err := <-svcErrors:
-		log.Event(ctx, "service error received", log.ERROR, log.Error(err))
+		log.Error(ctx, "service error received", err)
+		svc.Close(ctx)
+		return err
```
Comment (ChrysmOre) [inline main.go:59]: Here too?

File: service/service.go [hunk 3]
```diff
@@ -2,162 +2,203 @@ package service
...
+}
 
-	// Get Kafka consumer
-	consumer, err := serviceList.GetKafkaConsumer(ctx, cfg)
+// GetHealthCheck returns a healthcheck
+var GetHealthCheck = func(cfg *config.Config, buildTime, gitCommit, version string) (HealthChecker, error) {
+	versionInfo, err := healthcheck.NewVersionInfo(buildTime, gitCommit, version)
 	if err != nil {
-		log.Event(ctx, "failed to initialise kafka consumer", log.FATAL, log.Error(err))
 		return nil, err
```
Comment (ChrysmOre) [inline service/service.go:58]: Again, probably good to add context to the error here. Will add to cantabular-dataset service also

File: service/service.go [hunk 4]
```diff
@@ -2,162 +2,203 @@ package service
...
 
 	go func() {
 		defer cancel()
-		var hasShutdownError bool
 
 		// stop healthcheck, as it depends on everything else
-		if svc.serviceList.HealthCheck {
+		if svc.healthCheck != nil {
 			svc.healthCheck.Stop()
+			log.Event(ctx, "stopped health checker", log.INFO)
```
Comment (ChrysmOre) [inline service/service.go:142]: Opportunity to use log.Info rather than log.Event

**Review by ChrysmOre — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (DavidSubiros) [reply main.go:44]: Yes, please, I will change this one, if you could chagne cantabular-dataset that would be great

File: service/service.go (hunk 4 above)
Comment (DavidSubiros) [reply service/service.go:142]: thanks! I thought I had replaced all of them

File: service/service.go (hunk 3 above)
Comment (DavidSubiros) [reply service/service.go:58]: also the message is wrong!

Comment (DavidSubiros) [reply service/service.go:58]: I think this one is sorted (this is the old code) - for the new code we simple return the error

**Review by ChrysmOre — APPROVED** [review]

**Review by ChrysmOre — APPROVED** [review]

**Pull request merged**